doc/source/whatsnew/v1.4.0.rst

```diff
@@ -129,6 +129,7 @@ Other enhancements
 - :meth:`DataFrame.__pos__`, :meth:`DataFrame.__neg__` now retain ``ExtensionDtype`` dtypes (:issue:`43883`)
 - The error raised when an optional dependency can't be imported now includes the original exception, for easier investigation (:issue:`43882`)
 - Added :meth:`.ExponentialMovingWindow.sum` (:issue:`13297`)
+- :func:`guess_datetime_format` now correctly guesses the format of datetime strings with day of week and AM/PM placeholders (:issue:`43901`)
```

Comment:
this is not public, day ``` :func:`to_datetime` ``` and indicate what this does (speed it up?). this should go under datetime section